model/modeldecoder/v2/model.go

```diff
@@ -93,6 +98,51 @@ type context struct {
 	User user `json:"user"`
 }
 
+type contextFAAS struct {
+	// Indicates whether a function invocation was a cold start or not.
+	Coldstart nullable.Bool `json:"coldstart"`
+	// The request id of the function invocation.
+	Execution nullable.String `json:"execution"`
+	// The trigger type.
+	TriggerType nullable.String `json:"trigger.type"`
+	// The id of the origin trigger request.
+	TriggerRequestID nullable.String `json:"trigger.request_id"`
+}
+
+type contextCloud struct {
+	// Origin contains the self-nested field groups for cloud.
+	Origin contextCloudOrigin `json:"origin"`
+}
+
+type contextCloudOrigin struct {
+	// The cloud account or organization id used to identify
+	// different entities in a multi-tenant environment.
+	AccountID nullable.String `json:"account.id"`
+	// The cloud account name or alias used to identify different entities
+	// in a multi-tenant environment.
+	AccountName nullable.String `json:"account.name"`
+	// Availability zone in which this host, resource, or service is
+	// located.
+	AvailabilityZone nullable.String `json:"availability_zone"`
+	// Instance ID of the host machine.
+	InstanceID nullable.String `json:"instance.id"`
+	// Instance name of the host machine.
+	InstanceName nullable.String `json:"instance.name"`
+	// Machine type of the host machine.
+	MachineType nullable.String `json:"machine.type"`
+	// The cloud project name.
+	ProjectName nullable.String `json:"project.name"`
+	// The cloud project identifier.
+	ProjectID nullable.String `json:"project.id"`
+	// Name of the cloud provider.
+	Provider nullable.String `json:"provider"`
+	// Region in which this host, resource, or service is located.
+	Region nullable.String `json:"region"`
+	// The cloud service name is intended to distinguish services running
+	// on different platforms within a provider.
+	ServiceName nullable.String `json:"service.name"`
+}
```

Comment:
I'm not sure that all of these make sense. For example, would we ever know the machine type or instance ID of the origin? https://github.com/elastic/apm/pull/470/files proposes just the following:

- `cloud.origin.provider`
- `cloud.origin.region`
- `cloud.origin.service.name`
- `cloud.origin.account.id`

I would prefer to start out with only those that we need, and expand as necessary.

Reply:
That's fine by me. When I asked @AlexanderWert he said to limit it to all service.* and cloud.* fields available in ECS, so I added what was there.

Reply:
Sorry, my fault, was not aware we are already limiting the `cloud.*` and `service.*` intake fields to a subset of ECS. 
I'm also fine with @axw 's  proposal for `cloud.origin.*` and `service.origin.*`.

From the [AWS Lambda instrumentation spec PR](https://github.com/elastic/apm/blob/b98e151e4379f17754a393cb080875f0ca36e1b4/specs/agents/tracing-instrumentation-aws-lambda.md) the following *.origin.* fields are relevant:

- `service.origin.name`
- `service.origin.version`
- `service.origin.id`
- `cloud.origin.provider`
- `cloud.origin.region`
- `cloud.origin.service.name`
- `cloud.origin.account.id`